### ISSUE TITLE
Product Recommendation System

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/ML-Crate#20]
### ML-Crate Repository (Proposing new issue)
:red_circle: **Project Title** : Product Recommendation System
:red_circle: **Aim** : Build a product recommendation system like Amazon. A recommendation system can suggest you products, movies, etc based on your interests and the things you like and have used earlier.
:red_circle: **Dataset** : https://cseweb.ucsd.edu/~jmcauley/datasets.html
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

Hello, ML-Crate contributors, this issue is only for the contribution purposes and allocated only to the participants of SWOC 2.0 Open Source Program.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.
* A Demo Project has been created to guide you about the whole structure of presenting the Project in this repository. Here's the link of the Demo Project - https://github.com/abhisheks008/ML-Crate/tree/main/Project-Demo-Folder

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.
- This issue is only for 'SWOC' contributors of 'ML-Crate' project.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Participant ID :
- Approach for this Project :
- Are you a participant of SWOC 2.0?
    - [ ] YES
    - [ ]  No

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by Revanth-shivakumar** [comment]
Full name : Revanth K S
GitHub Profile Link : https://github.com/Revanth-shivakumar
Participant ID : 762
Approach for this Project: Perform Exploratory Data Analysis on the dataset and perform classification of user interests using clustering and grouping algorithmns and compare the accuracies to get the best model.
Are you a participant in SWOC 2.0? YES

**Comment by hpnightowl** [comment]
@Revanth-shivakumar  issue assigned to you. go ahead 🎉

**Comment by abhisheks008** [comment]
@Revanth-shivakumar It's been 25 days, what's the update here. Reply ASAP.

**Comment by demigod-22** [comment]
Full Name : M Arka Rutvik
Github profile : https://github.com/demigod-22
Participant ID : NA
Approach for this project : Perform exploratory data analysis and use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores.
Are you a participant of SWOC 2.0 - No
I wish to participate in KWOC 2022 and would like to contribute to this project.

**Comment by abhisheks008** [comment]
Issue assigned to you @demigod-22

**Comment by abhisheks008** [comment]
Unassigned. **ML-Crate** is going to be part of DWOC from Jan, 15, all the unfinished issues will be taken by the contributors of DWOC.

**Comment by milanprajapati571** [comment]
**Full name**: Milan Prajapati

**GitHub Profile Link**: [GitHub_Profile](https://github.com/milanprajapati571)

**Participant ID (If not, then put NA)**: NA

**Approach for this Project**:


1. Data Collection: Data Resource - the data set you provided.
2. Data Processing : 
      > Cleaning - Handle missing values, correct data types, etc. 
      > Normalization -  Scale numerical features. 
      > Encoding: Convert categorical data into numerical format.
3. Exploratory Data Analysis (EDA)
4. Model Selection:
      > Content-Based Filtering - Recommend items similar to those the user liked in the past.
      > Collaborative Filtering - Recommend items based on what similar users liked.
      > Hybrid Models - Combine both content-based and collaborative filtering.
5. Training Model:
      > Prepare DataLoader - For batch processing.
      > Training Loop - Forward pass, loss calculation, backward pass, optimizer step.
6. Model Evaluation:
      > RMSE (Root Mean Squared Error) - To measure accuracy.
      > Precision, Recall, F1-Score - For classification-based evaluation.
7. Making recommendations: generating the prediction from the trained model.
8. Documentation: noting down the very small steps of the whole journey of making the project in a single file.

**What is your participant role?** : SSoC (Social Summer of Code)

Sir, can You Please assign this project to me...?

**Comment by abhisheks008** [comment]
Implement as much models as you can for this problem statement.

Assigned @milanprajapati571
